src/AppInstallerCLICore/Workflows/UpdateFlow.cpp

```diff
@@ -0,0 +1,185 @@
+                context.Add<Execution::Data::Installer>(std::move(installer));
+
+                updateFound = true;
+            }
+            else
+            {
+                // Any following versions are not applicable
+                break;
+            }
+        }
+
+        if (!updateFound)
+        {
+            context.Reporter.Info() << Resource::String::UpdateNotApplicable << std::endl;
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_UPDATE_NOT_APPLICABLE);
+        }
+    }
+
+    void EnsureUpdateVersionApplicable(Execution::Context& context)
+    {
+        Utility::Version updateVersion(context.Get<Execution::Data::Manifest>().Version);
+
+        if (!IsUpdateVersionApplicable(context, updateVersion))
+        {
+            context.Reporter.Info() << Resource::String::UpdateNotApplicable << std::endl;
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_UPDATE_NOT_APPLICABLE);
+        }
+    }
+
+    void GetUpdateManifestAndInstaller(Execution::Context& context)
+    {
+        if (context.Args.Contains(Execution::Args::Type::Manifest))
+        {
+            context <<
+                GetManifestFromArg <<
+                ReportManifestIdentity <<
+                GetCompositeSourceFromInstalledAndAvailable <<
+                SearchSourceUsingManifest <<
+                EnsureOneMatchFromSearchResult <<
+                GetInstalledPackage <<
+                EnsureUpdateVersionApplicable <<
+                EnsureMinOSVersion <<
+                SelectInstaller <<
+                EnsureApplicableInstaller;
+        }
+        else
+        {
+            context <<
+                GetCompositeSourceFromInstalledAndAvailable <<
+                SearchSourceForSingle <<
+                EnsureOneMatchFromSearchResult <<
+                ReportSearchResultIdentity <<
+                GetInstalledPackage <<
+                GetUpdateManifestAndInstallerFromSearchResult;
+        }
+    }
+
+    void UpdateAllApplicable(Execution::Context& context)
+    {
+        const auto& matches = context.Get<Execution::Data::SearchResult>().Matches;
+        bool updateAllHasFailure = false;
+        for (const auto& match : matches)
+        {
+            std::thread tryOneUpdate([&]
+                {
+                    // We want to do best effort to update all applicable updates regardless on previous update failure
+                    context.Resume();
+                    context.Reporter.Info() << std::endl;
+
+                    context.Add<Execution::Data::InstalledPackageVersion>(match.Package->GetInstalledVersion());
+
+                    context <<
+                        SelectLatestApplicableUpdate(*(match.Package)) <<
+                        ShowInstallationDisclaimer <<
+                        DownloadInstaller <<
+                        ExecuteInstaller <<
+                        RemoveInstaller;
+                });
+
+            tryOneUpdate.join();

```

Comment:
>tryOneUpdate.join(); [](start = 12, length = 20)

Why are you running this on a separate thread just to immediately wait for it?

Reply:
AICLI_Terminate() has return; which will exit the current method immediately.

---
In reply to: [499935455](https://github.com/microsoft/winget-cli/pull/588#discussion_r499935455) [](ancestors = 499935455)

Reply:
I can put these in a separate method 

---
In reply to: [499936643](https://github.com/microsoft/winget-cli/pull/588#discussion_r499936643) [](ancestors = 499936643,499935455)

Reply:
Even with a forced return, you can make it a lambda or named function.

---
In reply to: [499938571](https://github.com/microsoft/winget-cli/pull/588#discussion_r499938571) [](ancestors = 499938571,499936643,499935455)